Supports more cookie date formats in the Cookie Plugin

aaa2000 opened the pull request:
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes #46
| License         | MIT


#### What's in this PR?

Supports more cookie date formats in the Cookie Plugin

#### Checklist

- [x] Updated CHANGELOG.md to describe BC breaks / deprecations | new feature | bugfix
- [ ] Documentation pull request created (if not simply a bugfix)


#### To Do

- [x] Wait the release 1.6 of php-http/message in order to fix the composer constraint to a stable constraint

Nyholm commented on composer.json:
```diff
@@ -14,7 +14,7 @@
         "php-http/httplug": "^1.1",
         "php-http/message-factory": "^1.0",
-        "php-http/message": "^1.2",
+        "php-http/message": "^1.6@dev",
```
You should remove "@dev"

Possible add "minimun-stabillity:dev" and "prefer-dist "

dbu replied:
see the TODO list @aaa2000 added. i think this is correct to show that the PR will merge once 1.6 is released. its just that we should not merge until 1.6 is released and then we can change this line to remove `@dev` before merging.

changing global minimum stability will not change anything for users of this lib, but would potentially have effects on how we install other dependencies.

Nyholm replied:
I missed that. I think we could release 1.6 of message now.. I will prepare.

tcitworld commented:
Could this be released ? :)

aaa2000 commented:
Rebased and fixed the composer constraint to a stable constraint for the package `php-http/message`

PR is now mergeable

Do I need to correct StyleCI analysis ? (I have not modified `src/Plugin/CookiePlugin.php` )

Nyholm commented:
I think you are fine. I will review this PR later this week. 

Thank you.

dbu commented:
> Do I need to correct StyleCI analysis

i let styleci fix those in master. if you can rebase again, the style check should become green.

aaa2000 commented:
Friendly ping

joelwurtz commented:
Thanks for the PR, and sorry for the wait

aaa2000 commented:
Thanks to you and your wonderful library
